### PR TITLE
Add the regions filter and fix some old data

### LINKED ISSUE #1077: Adding Filter by regions (e.g., North America, Europe, Asia, etc.)
This feature will allow users to filter conferences based on their geographical regions, enabling easier navigation and more targeted event searches. Implementing this filter will help users quickly find conferences relevant to their location or regions of interest.

## Steps to Reproduce:
- Navigate to the developer conferences agenda table.
- Observe the absence of a region-based filter option.

## Expected Behavior:

- Users should be able to filter the conferences displayed in the agenda table by selecting a specific region. This will refine the list to show only those conferences that are held in the selected region.

## Actual Behavior:
- Currently, there is no option to filter conferences by regions, making it less convenient for users looking for geographically relevant events.

## Proposed Solution:
- Add a dropdown menu or a set of checkboxes for regions (e.g., North America, Europe, Asia, etc.) in the agenda table interface.
- Implement the backend logic to filter the displayed conferences based on the selected regions.
- Ensure that the filtering functionality is smooth and does not impact the performance of the agenda table.

### PR DESCRIPTION
**Pull request by jespino**
Fixes #1077

### CODE REVIEW

**Comment by scraly** [comment]
Thanks :)
A first remark: the new "validateRegions.js" script is missing in run.sh

**Comment by scraly** [comment]
I tested it and I saw a bug

If we select a region, all of the countries are still displayed in the list, so we can select a region with a country that is not in this country, example:

image


A suggestion may be to hide the country list if the region is selected. And if the region is equals to All, then the country list is displayed again?

File: .github/workflows/check_regions.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+name: CheckRegions
+on: pull_request
+
+jobs:
+  MdValidator:
```
Comment (scraly) [inline .github/workflows/check_regions.yml:5]:
```suggestion
  RegionsValidator:
```

**Review by scraly — CHANGES_REQUESTED** [review]

**Comment by jespino** [comment]
@scraly I update it to filter the countries and only list the ones that are inside the region, and whenever you change the region I automatically deselect the country.

**Comment by scraly** [comment]
Thanks
I observed a new issue:
in the first load, when you click in the filter, the country list is hidden, we should see the list:
image

Users should be allowed to select a country if they want (to display events in Spain for example :) )

**Comment by jespino** [comment]
@scraly that was an easy one. Fixed.

**Comment by scraly** [comment]
Thanks ^^
Easy fix and useful :) 

I see another thing, in the list of countries (after selecting a region), is it possible to sort the countries alphabetically?

**Comment by jespino** [comment]
@scraly done

File: page/src/misc/regions.json [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+  "Asia": ["Russia", "India", "China", "Israel", "Turkey", "Japan", "Thailand", "Armenia", "Bangladesh", "Dubai", "Georgia", "Hong Kong", "United Arab Emirates", "Vietnam", "Pakistan", "Jordan", "Laos", "Lebanon", "Nepal", "Philippines", "Singapore", "South Korea", "Sri Lanka", "Taiwan"],
+  "Central America": ["Costa Rica", "Nicaragua", "Cayman Islands", "Dominican Republic", "El Salvador", "Honduras", "Trinidad"],
+  "Europe": ["Spain", "France", "UK", "Germany", "Italy", "Greece", "Portugal", "Poland", "Sweden", "Belgium", "Netherlands", "Switzerland", "Finland", "Norway", "Estonia", "Denmark", "Austria", "Czech Republic", "Croatia", "Romania", "Latvia", "Belarus", "Bosnia and Herzegovina", "Bulgaria", "Kosovo", "Hungary", "Iceland", "Ireland", "Slovakia", "Slovenia", "Wales", "Scotland", "Serbia", "Lithuania", "Luxembourg", "Macedonia", "Malta", "Ukraine", "Martinique", "North Macedonia"],
+  "North America": ["USA", "Canada", "Mexico"],
+  "Oceania": ["Australia", "New Zealand", "Indonesia"],
+  "South America": ["Argentina", "Brazil", "Chile", "Peru", "Colombia", "Bolivia", "Columbia"]
```
Comment (scraly) [inline page/src/misc/regions.json:8]:
```suggestion
  "South America": ["Argentina", "Brazil", "Chile", "Peru", "Colombia", "Bolivia"]
```

**Review by scraly — CHANGES_REQUESTED** [review]

**Review by scraly — APPROVED** [review]
approved

**Pull request merged**